File: CreditCardGenerator(8809)/frontend/CardGenerator/src/app/services/credit-card/credit-card.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/internal/Observable';


interface CustomerDto{
  name:string;
  bankname:string;
}

export interface CreditCardDto{
  cardNumber:string;
  expiryDate : Date;
  cvv:number;
  name:string;
  bankName:string;
}

const BASIC_URL = "http://localhost:8809/card";

@Injectable({
  providedIn: 'root'
})
export class CreditCardService {

  constructor(private http: HttpClient) { }

  generateCreditCard(customer : CustomerDto) : Observable<any>{

    return this.http.post<CreditCardDto>(BASIC_URL+`/generateCreditCard` , customer);

  }


  getCardDetails(name:string) : Observable<CreditCardDto>{

    return this.http.get<CreditCardDto>(BASIC_URL + `/getCardDetails/${name}`);

  }

  getAllCards() : Observable<CreditCardDto[]> {

    return this.http.get<CreditCardDto[]>(BASIC_URL + `/getAllCards`);

  }

  
}

File: CreditCardGenerator(8809)/backend/system/src/main/java/com/example/system/service/CreditCardService.java
package com.example.system.service;

import com.example.system.dto.CreditCardDto;
import com.example.system.dto.CustomerDto;
import com.example.system.model.CreditCardEntity;

import java.util.List;

public interface CreditCardService {

    CreditCardDto generateCreditCard(CustomerDto customerDto) throws IllegalAccessException;

    CreditCardEntity getCreditCardDetailsByName(String name);

    List<CreditCardEntity> getAllCards();

}
